cmd/object_delete.go

```diff
@@ -0,0 +1,88 @@
+/***************************************************************
+*
+* Copyright (C) 2024, Pelican Project, Morgridge Institute for Research
+*
+* Licensed under the Apache License, Version 2.0 (the "License"); you
+* may not use this file except in compliance with the License.  You may
+* obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*
+***************************************************************/
+
+package main
+
+import (
+	"fmt"
+	"os"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+
+	"github.com/pelicanplatform/pelican/client"
+	"github.com/pelicanplatform/pelican/config"
+)
+
+var (
+	objectDeleteCmd = &cobra.Command{
+		Use:   "delete {object}",
+		Short: "Delete an object or a collection",
+		Args: func(cmd *cobra.Command, args []string) error {
+			if len(args) < 1 {
+				return fmt.Errorf("no location provided for deletion")
+			}
+			if len(args) > 1 {
+				return fmt.Errorf("too many arguments provided; only one argument is allowed")
+			}
+			return nil
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			log.Warning("Object delete command is not openly supported and will only remove data from the origin, not from caches the data might have propagated to.")
```

Comment:
Let's also mention "from the origin pointed to by the Director's collections url" to be clear that multi-origin namespaces aren't yet fully supported.

Reply:
With the recent changes, the delete command can also use the first origin URL it finds if the collection URL is not present. So, I think I'll be better to not go ahead with this change.